src/sql/base/browser/ui/chart/chart.component.ts

```diff
@@ -0,0 +1,321 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+import { Component, Inject, forwardRef, ChangeDetectorRef } from '@angular/core';
+import * as chartjs from 'chart.js';
+import { mixin } from 'sql/base/common/objects';
+import { Disposable } from 'vs/base/common/lifecycle';
+import * as azdata from 'azdata';
+
+export interface BarDataSet {
+	label: string;
+	data: number[];
+	backgroundColor?: string[];
+	borderColor?: string[];
+}
+
+@Component({
+	selector: 'chart-component',
+	templateUrl: decodeURI(require.toUrl('./chart.component.html'))
+})
+export class Chart<TConfig extends azdata.ChartConfiguration> extends Disposable {
+	private _chartId: string;
+	private _type: azdata.ChartType;
+	private _configuration: chartjs.ChartData;
+
+	public chart: chartjs.Chart;
+	private canvas: HTMLCanvasElement;
+	public element: string;
+
+	/**
+	 * Options in the form that Chart.js accepts (hence the `any` type)
+	 */
+	private _options: any = {
+		events: ['click', 'keyup'],
+		responsive: true,
+		maintainAspectRatio: false
+	};
+
+	constructor(
+		@Inject(forwardRef(() => ChangeDetectorRef)) private _changeRef: ChangeDetectorRef
+	) {
+		chartjs.Chart.register(...chartjs.registerables);
+		super();
+	}
+
+	ngAfterViewInit(): void {
+
+	}
+
+	/**
+	 * Setter function for chart type
+	 */
+	public set type(val: any) {
+		if (val === 'horizontalBar') {
+			this._type = 'bar';
+			this._options = mixin({}, mixin(this._options, { indexAxis: 'y' }));
+		}
+		else {
+			this._type = val;
+		}
+		this._changeRef.detectChanges();
+	}
+
+	/**
+	 * Setter function for chart configuration
+	 */
+	public set configuration(val: TConfig) {
+		this._configuration = this.convert(val);
+
+		if ((<any>val).options) { // TODO: give TConfig a strongly-typed TOptions param
```

Comment:
TODO is out-of-date.  I'm exploring whether/how to separate out options from data so that they can be updated separately, while maintain correct typing.